main.tf

```diff
@@ -197,13 +204,106 @@ data "aws_iam_policy_document" "lambda_access" {
   }
 }
 
+data "aws_iam_policy_document" "organizations_readonly_access" {
+  count = module.this.enabled && length(var.organizations_readonly_access) > 0 ? 1 : 0
+
+  statement {
+    sid    = "ReadonlyAccess"
+    effect = "Allow"
+
+    principals {
+      identifiers = ["*"]
+      type        = "*"
+    }
+
+    condition {
+      test     = "StringEquals"
+      values   = [var.organizations_readonly_access]
+      variable = "aws:PrincipalOrgID"
+    }
+
+    actions = [
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:BatchGetImage",
+      "ecr:DescribeImageScanFindings",
+      "ecr:DescribeImages",
+      "ecr:DescribeRepositories",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:GetLifecyclePolicy",
+      "ecr:GetLifecyclePolicyPreview",
+      "ecr:GetRepositoryPolicy",
+      "ecr:ListImages",
+      "ecr:ListTagsForResource",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "organization_full_access" {
+  count = module.this.enabled && length(var.organizations_full_access) > 0 ? 1 : 0
+
+  statement {
+    sid    = "ReadonlyAccess"
+    effect = "Allow"
+
+    principals {
+      identifiers = ["*"]
+      type        = "*"
+    }
+
+    condition {
+      test     = "StringEquals"
+      values   = [var.organizations_readonly_access]
+      variable = "aws:PrincipalOrgID"
+    }
+
+    actions = [
+      "ecr:*",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "organization_push_access" {
+  count = module.this.enabled && length(var.organizations_push_access) > 0 ? 1 : 0
+
+  statement {
+    sid    = "ReadonlyAccess"
+    effect = "Allow"
+
+    principals {
+      identifiers = ["*"]
+      type        = "*"
+    }
+
+    condition {
+      test     = "StringEquals"
+      values   = [var.organizations_push_access]
+      variable = "aws:PrincipalOrgID"
+    }
+
+    actions = [
+      "ecr:CompleteLayerUpload",
+      "ecr:GetAuthorizationToken",
+      "ecr:UploadLayerPart",
+      "ecr:InitiateLayerUpload",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:PutImage",
+    ]
+  }
+}
+
 data "aws_iam_policy_document" "resource" {
   count                   = module.this.enabled ? 1 : 0
-  source_policy_documents = local.principals_readonly_access_non_empty ? [data.aws_iam_policy_document.resource_readonly_access[0].json] : [data.aws_iam_policy_document.empty[0].json]
+  source_policy_documents = local.principals_readonly_access_non_empty ? [
```

Comment:
I don't think this change is right, as it doesn't allow to provide both principals and organizations with read access for the same ECR repository. Maybe revert this change, and put the organizations read logic in the override list?